RTClib.cpp

```diff
@@ -1519,6 +1519,164 @@ void RTC_PCF8563::writeSqwPinMode(Pcf8563SqwPinMode mode) {
   Wire._I2C_WRITE(mode);
   Wire.endTransmission();
 }
+
+/**************************************************************************/
+/*!
+    @author Leonardo Bispo
+    @brief  Set the Alarm values mode on the PCF8563
+    @details  One can only set day, hour and minutes for the alarm, the other
+      values of DateTime are ignored, but provided to comply with the format.
+    @param dt DateTime to set
+*/
+/**************************************************************************/
+void RTC_PCF8563::setAlarm(const DateTime &dt, Pcf8563AlarmMode alarm_mode) {
+
+  uint8_t minute = bin2bcd(dt.minute());
+  uint8_t hour = bin2bcd(dt.hour());
+  uint8_t day =
+      (alarm_mode == PCF8563_Alarm_Monthly) ? bin2bcd(dt.day()) : 0x80;
+  uint8_t weekDay =
+      (alarm_mode == PCF8563_Alarm_Weekly) ? bin2bcd(dt.dayOfTheWeek()) : 0x80;
+
+  if (alarm_mode == PCF8563_Alarm_hourly) {
+    hour = 0x80;
+  }
+
+  write_i2c_register(PCF8563_ADDRESS, PCF8563_MINUTE_ALARM, minute);
+  write_i2c_register(PCF8563_ADDRESS, PCF8563_HOUR_ALARM, hour);
+  write_i2c_register(PCF8563_ADDRESS, PCF8563_DAY_ALARM, day);
+  write_i2c_register(PCF8563_ADDRESS, PCF8563_WEEKDAY_ALARM, weekDay);
+
+  uint8_t control_status_2 =
+      read_i2c_register(PCF8563_ADDRESS, PCF8563_CONTROL_2);
+
+  // Enable alarm interrupt AIE
+  write_i2c_register(PCF8563_ADDRESS, PCF8563_CONTROL_2,
+                     (control_status_2 | 0x02)); // 00000010b
+}
+
+/**************************************************************************/
+/*!
+    @author Leonardo Bispo
+    @brief  Set Weekly Alarm values mode on the PCF8563
+    @param dow  Day of the week to trigger the alarm
+    @param hour Hour to trigger the alarm
+    @param min  Minute to trigger the alarm
+*/
+/**************************************************************************/
+void RTC_PCF8563::setAlarm(DayOfWeek dow, uint8_t hour, uint8_t min) {
+
+  uint32_t t = SECONDS_FROM_1970_TO_2000;
+  switch (dow) {
+  case Sundays:
+    t = 1483228800; // Sunday, January 1, 2017 0:00:00
+    break;
+  case Mondays:
+    t = 1483315200;
+    break;
+  case Tuedays:
+    t = 1483401600;
+    break;
+  case Wednesdays:
+    t = 1483488000;
+    break;
+  case Thursdays:
+    t = 1483574400;
+    break;
+  case Fridays:
+    t = 1483660800;
+    break;
+  case Saturdays:
+    t = 1483747200; // Saturday, January 7, 2017 0:00:00
+    break;
+  }
+
+  setAlarm(DateTime(t), PCF8563_Alarm_Weekly);
```

Comment:
This `DateTime` constructor is computationally expensive. It is much cheaper to:
```c++
setAlarm(DateTime(2017, 1, 1 + dow, hour, min), PCF8563_Alarm_Weekly);
```